V1Michigan/v1
# Add basic Supabase login

### Status
**READY**

### Description
Sets up basic Supabase sign-in with Google (`/login`) and an user profile/account page (`/account`). Next steps will be to add email/password sign-in and a more complete sign-in flow and add more info to the profile page.

Also adds `/welcome`, currently unused.


### Dependencies
- Added `@supabase/supabase-js`
- Added `react-spinners`
- Added `react-tsparticles`

### Migrations
NO

### Steps to Test or Reproduce

```sh
git pull --prune
git checkout supalogin
npm run dev
```

Also make sure you have the updated `.env` with Supabase keys (in Slack).

Note that redirecting won't work on the Netlify build preview (unless we make it an allowed redirect in Supabase).

1. Test that `localhost:3000/login` sign-in with Google works as expected and redirects to `/account`
2. Test that `/login` redirects to `/account` when signed in
3. Test that `/account` redirects to `/login` when not signed in

`/account` is still broken and will raise an alert, but @mdvsh's work should fix this eventually.

### Impacted Areas in Application

* Nothing currently visible to visitors, just adds the new `/account` and `/login` routes
